### ISSUE TITLE
Azure Front Door TLS Version Support

### ISSUE THREAD

**Issue by BenTheCloudGuy** [terraform-providers/terraform-provider-azurerm#4434]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

  Azure Front Door API now supports passing in a minimum TLS version (either 1.0 or 1.2) as part of the custom https configuration. I would like to propose adding support for this feature to the Azure Front Door resource. 

### New or Affected Resource(s)

* azurerm_frontdoor

### Potential Terraform Configuration

```hcl
resource "azurerm_frontdoor" "example" {
  name                                         = "example-FrontDoor"
  location                                     = "${azurerm_resource_group.example.location}"
  resource_group_name                          = "${azurerm_resource_group.example.name}"
  enforce_backend_pools_certificate_name_check = false

  routing_rule {
      name                    = "exampleRoutingRule1"
      accepted_protocols      = ["Http", "Https"]
      patterns_to_match       = ["/*"]
      frontend_endpoints      = ["exampleFrontendEndpoint1"]
      forwarding_configuration {
        forwarding_protocol   = "MatchRequest"
        backend_pool_name     = "exampleBackendBing"
      }
  }

  backend_pool_load_balancing {
    name = "exampleLoadBalancingSettings1"
  }

  backend_pool_health_probe {
    name = "exampleHealthProbeSetting1"
  }

  backend_pool {
      name            = "exampleBackendBing"
      backend {
          host_header = "www.bing.com"
          address     = "www.bing.com"
          http_port   = 80
          https_port  = 443
      }

      load_balancing_name = "exampleLoadBalancingSettings1"
      health_probe_name   = "exampleHealthProbeSetting1"
  }

  frontend_endpoint {
    name                              = "exampleFrontendEndpoint1"
    host_name                         = "example-FrontDoor.azurefd.net"
    custom_https_provisioning_enabled = false
    custom_https_configuration {
        minimum_tls_version           = "1.2"
        certificate_source            = "FrontDoor"
    }
  }
}
```

### References

```json
"minimumTlsVersion": {
          "description": "The minimum TLS version required from the clients to establish an SSL handshake with Front Door.",
          "enum": [
            "1.0",
            "1.2"
          ],
          "type": "string",
          "x-ms-enum": {
            "name": "MinimumTLSVersion",
            "modelAsString": true
          }
        },
```

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-05-01/frontdoor.json

**Comment by mybayern1974** [comment]
@WodansSon for awareness.

**Comment by WodansSon** [comment]
@BenTheCloudGuy This feature isn't currently supported in the 2019-04-01 version of the API, I will pair this fix with the fix for #4558 since they both require the version of the API to be changed.

**Comment by WodansSon** [comment]
The support for this feature shipped in the v1.43 release, see resource [documentation](https://www.terraform.io/docs/providers/azurerm/r/front_door.html#minimum_tls_version)
